File: golang/singleton.go
package main

import "fmt"

type Logger struct {
	name string
}

var logger *Logger

func GetInstance() *Logger {
	if logger == nil {
		logger = &Logger{name: "Logger"}
		fmt.Println("creating new instance")
	} else {
		fmt.Println("instance already exists")
	}
	return logger
}

func main() {
	firstLogger := GetInstance()
	secondLogger := GetInstance()

	if firstLogger == secondLogger {
		fmt.Println("both are same loggers")
	}
}

File: python/singleton.py
class Logger(object):
    _instance = None

    def __new__(cls):
        if cls._instance is None:
            print("creating new instance")
            cls._instance = super(Logger, cls).__new__(cls)
        else:
            print("instance already exists")
        return cls._instance


first_logger = Logger()
second_logger = Logger()

if first_logger == second_logger:
    print("both are same loggers")
